kbfsgit/runner.go

```diff
@@ -357,11 +360,59 @@ func humanizeBytes(n int64, d int64) string {
 	return fmt.Sprintf("%.2f/%.2f GB", float64(n)/gbf, float64(d)/gbf)
 }
 
+func (r *runner) printStageStart(ctx context.Context,
+	toPrint []byte, memProfName, cpuProfName string) {
+	if len(toPrint) == 0 {
+		return
+	}
+	r.printStageEndMaybe(ctx)
+
+	r.printStageLock.Lock()
+	defer r.printStageLock.Unlock()
+
+	r.stageStartTime = r.config.Clock().Now()
+	r.stageMemProfName = memProfName
+
+	if len(cpuProfName) > 0 && r.verbosity >= 4 {
+		cpuProfPath := filepath.Join(
+			os.TempDir(), cpuProfName)
+		f, err := os.Create(cpuProfPath)
+		if err != nil {
+			r.log.CDebugf(
+				ctx, "Couldn't create CPU profile: %s", cpuProfName)
+			cpuProfPath = ""
+		} else {
+			pprof.StartCPUProfile(f)
+		}
+		r.stageCPUProfPath = cpuProfPath
+	}
+
+	r.errput.Write(toPrint)
+	r.needPrintDone = true
+}
+
+// printStageEndMaybe should only be used to end stages started with
+// printStageStart.
+func (r *runner) printStageEndMaybe(ctx context.Context) {
```

Comment:
Please move up above its first use.

Reply:
Also, how about `printStageEndIfNeeded()`?  That's a little more clear to me.

Reply:
Hmm I think it's already above every use of it?

Reply:
Yup this is above `printStageStart` :)

Reply:
Am I going crazy?
`printStageStart`, line 363: https://github.com/keybase/kbfs/pull/1221/files/c11d716a6163d1267380e6bd992876559905e6d5#diff-b6999c82a099222a06e9532d0ff7e054R363
`printStageEndMaybe`, line 396: https://github.com/keybase/kbfs/pull/1221/files/c11d716a6163d1267380e6bd992876559905e6d5#diff-b6999c82a099222a06e9532d0ff7e054R396

Reply:
Oh, looks like your last "better name" commit moved it.

Reply:
No I moved it before the `better name`! Hmm unless I forgot to save before commit "review feedback"?

Reply:
Ugh looks like so. Sorry!